content/en/docs/reference/kubernetes-api/ref/common-definitions/podaffinity-.md

```diff
@@ -0,0 +1,73 @@
+---
+api_metadata:
+  apiVersion: ""
+  import: "k8s.io/api/core/v1"
+  kind: "PodAffinity"
+collapsible: false
+content_type: "api_reference"
+description: "Pod affinity is a group of inter pod affinity scheduling rules."
+draft: false
+title: "PodAffinity"
+weight: 16
+---
+
+
+
+`import "k8s.io/api/core/v1"`
+
+
+## PodAffinity
+
+Pod affinity is a group of inter pod affinity scheduling rules.
+
+
+- **preferredDuringSchedulingIgnoredDuringExecution** ([]WeightedPodAffinityTerm)
+
+  The scheduler will prefer to schedule pods to nodes that satisfy the affinity expressions specified by this field, but it may choose a node that violates one or more of the expressions. The node that is most preferred is the one with the greatest sum of weights, i.e. for each node that meets all of the scheduling requirements (resource request, requiredDuringScheduling affinity expressions, etc.), compute a sum by iterating through the elements of this field and adding "weight" to the sum if the node has pods which matches the corresponding podAffinityTerm; the node(s) with the highest sum are the most preferred.
+
+  *The weights of all of the matched WeightedPodAffinityTerm fields are added per-node to find the most preferred node(s)*
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm** (PodAffinityTerm), required
```

Comment:
So as a reader, I will have to parse the JSON paths on line 36, 40 and 44 to understand what a `PodAffinityTerm` is.
Do we still believe inline expansion of definitions is a good idea?

Reply:
Yes, I agree that expanding the definitions on several levels without indentation is not helping the reader. I try to find a compromise between not indenting too much, having all the information in the same place and making the structure understandable.